### ISSUE TITLE
Ability to NOT redact certain Secret values

### ISSUE THREAD

**Issue by maiamcc** [windmilleng/tilt#2417]
from Tom at Yext:
> this is possibly an "operator error" kinda thing, but we've got a secret for a username in a cloud service, and the username is a generic word that appears naturally in the logs frequently, so we get a lot of redactions in paths, etc.
> since the secret name is included, this does expose the actual content of the secret, which isn't too big of a deal in this specific case.
> but would there be a way to exclude that particular word from the redaction?

proposed solution is a Tiltfile func `unredact_secret(name='username')` or similar.

**Comment by zkutasi** [comment]
I would just like to bump this up... could you at least make this redacting feature optional? it is the simplest I think... during development, I could not care less about redacting secrets. But this makes log-reading a pain.

**Comment by maiamcc** [comment]
@zkutasi good point, let me take a look, it might be pretty easy to disable scrubbing.

**Comment by victorwuky** [comment]
@zkutasi : This is now available in the latest release. See https://docs.tilt.dev/api.html#api.secret_settings and https://github.com/tilt-dev/tilt/releases/tag/v0.13.6

**Comment by whitehat101** [comment]
I think this feature request is still valid. The API mentioned `secret_settings(True)` turns off _all_ scrubbing. It would be nice to have control over what is scrubbed or not.

I'm loading a `.env` file as a secret, which should be a common pattern. It contains secrets, but also some non-secrets, like `LOG_LEVEL`. I'm seeing logs like:
```
[redacted secret env-secrets:LOG_LEVEL] [server:server.py:18] debug message
```

Which is silly.

I could:

1. use `secret_settings(True)`, and see passwords in the logs
2. load `.env` as a config map instead, and see passwords in the logs
3. setup two `.env` files, e.g.: `.env.secret` and  `.env.public` loading one as a secret and one as a config map

Option 3 is technically the most correct (and what we do in prod), but for devs it's more complex and more confusing. Lots of tooling will work with a `.env` file, but won't auto-load `.env.*` files.

It would be nice to have options like (at least one of) the following, that sets up allow lists for either secret values, secret keys, or secret-namespaced keys:

```
secret_settings(permit_values=["true", "false", "INFO", "DEBUG", ... ])
secret_settings(permit_keys=["LOG_LEVEL"])
secret_settings(permit_secrets=["env-secrets:LOG_LEVEL"])
```

Compiling a list of common non-secret secret values ("true", "false", "enabled", "yes", "password") and defaulting to allowing them, even if they are in a secret would be a nice user experience.
